src/components/Search/index.tsx

```diff
@@ -132,20 +159,34 @@ const Search: FC<{
     if (isClear) {
       setSearchValue('')
       clearSearchInput(inputElement)
+      setSearchByNameResults(null)
       onEditEnd?.()
     }
   }
 
-  const inputChangeAction = (event: ChangeEvent<HTMLInputElement>) => {
-    setSearchValue(event.target.value)
-    if (!event.target.value) onEditEnd?.()
+  const inputChangeAction = async (event: ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value
+    setSearchValue(inputValue)
+
+    if (isSearchByNames && inputValue) {
+      debouncedSearchByName(inputValue)
```

Comment:
I think `onChange` and `onKeyUp` can be purer, only handling `setKeyword` and `onEditEnd`.

Then use `useEffect` to listen for changes to `keyword` to trigger `searchById / searchByName`.

Reply:
here the business logic is more complex, while `searching by id` is enabled, trigger search only when user press `Enter` or click the search button

Reply:
It sounds like there should also be an `onEditEnd` internally, where `props.onEditEnd` and `searchById / searchByName` are triggered.

The `useEffect` should only skip when `searchByName` is `false`.

Reply:
The `onEditEnd` is used here: https://github.com/Magickbase/ckb-explorer-frontend/blob/develop/src/components/Header/SearchComp/index.tsx#L11-L14
```ts
  const onEditEnd = useCallback(() => {
    setExpanded?.(false)
    hideMobileMenu?.()
  }, [hideMobileMenu, setExpanded])
```

so that when search by name we don't need it

Reply:
The externally exposed `onEditEnd` and the internally implemented `onEditEnd` are two different things.

Whether `onEditEnd` is exposed externally does not affect whether an internal `onEditEnd` exists.

You can consider them as having the same name by coincidence.

> so that when search by name we don't need it

I didn't understand your intention here. Are you suggesting that you don't want the `setExpanded` logic to be triggered when pressing Enter in the search box under `by name`? However, I don't see such logic currently.

If you do want to implement this, you can consider abstracting a new externally exposed interface, such as `onSearchEnd`.

Reply:
> It sounds like there should also be an onEditEnd internally, where props.onEditEnd and searchById / searchByName are triggered.

what I tried to say is that there is no side effect need to be triggerred when search by name, so maybe I did't need an extra `onEditEnd`